src/icrc1_auction_api_mock.mo

```diff
@@ -330,26 +331,34 @@ actor class Icrc1AuctionAPI(adminPrincipal_ : ?Principal) = self {
     return null;
   };
 
-  public shared query ({ caller }) func queryTokenBids(ledger : Principal) : async [(Auction.OrderId, Order)] = async switch (getAssetId(ledger)) {
-    case (?aid) {
-      U.unwrapUninit(auction).getOrders(caller, #bid, ?aid)
-      |> Array.tabulate<(Auction.OrderId, Order)>(_.size(), func(i) = mapOrder(_ [i]));
+  public shared query ({ caller }) func queryTokenBids(ledger : Principal) : async ([(Auction.OrderId, Order)], Nat) {
+    let orders = switch (getAssetId(ledger)) {
+      case (?aid) {
+        U.unwrapUninit(auction).getOrders(caller, #bid, ?aid)
+        |> Array.tabulate<(Auction.OrderId, Order)>(_.size(), func(i) = mapOrder(_ [i]));
+      };
+      case (_) [];
     };
-    case (_) [];
+    (orders, U.unwrapUninit(auction).sessionsCounter);
```

Comment:
I don't really understand what `auction` is if `getAssetId(ledger)` returns null. How can that be defined?

Reply:
The auction can be working, but user provided unknown principal as ledger principal